content/modules/redisgears/jvm/quickstart.md

```diff
@@ -0,0 +1,259 @@
+	public Reviews(int count, int ratingsSum) {
+		this.count = count;
+		this.ratingsSum = ratingsSum;
+	}
+
+    public static void main(String args[]) 
+    {  
+        // Create the reader that will pass data to the pipe
+        KeysReader reader = new KeysReader();
+        
+        // Create the data pipe builder
+        GearsBuilder<KeysReaderRecord> gb = GearsBuilder.CreateGearsBuilder(reader);
+        
+		gb.filter(r->{
+			// Filter out any keys that are not reviews
+			return r.getKey().startsWith("review:");
+		}).map(r->{
+			// Extract the rating field
+			return r.getHashVal().get("rating");
+		})
+		.accumulate(new Reviews(0, 0), (accumulator, record)-> {
+			// Count the reviews and add up all of their ratings
+			accumulator.count++;
+			accumulator.ratingsSum += Integer.parseInt(record);
+			return accumulator;
+		}).map(r->{
+			// Calculate the average rating
+			return ((double) r.ratingsSum) / r.count;
+		});
+             
+        // Run the data through the pipeline immediately
+        gb.run();
+    }
+}
+```
+
+#### Example output
+
+```sh
+$ redis-cli -x -h {host} -p {port} \
+    RG.JEXECUTE com.domain.packagename.Reviews < /tmp/rgjvmtest-0.0.1-SNAPSHOT.jar
+1) 1) "3.6666666666666665"
+2) (empty array)
+```
+
+### Event processing
+
+If you use the `GearsBuilder.register()` function in your code, then the functions you add to your pipeline will run every time a certain database event occurs.
+
+The following example registers a pipeline of functions that will automatically update the maximum age every time you add a new person hash to your database.
+
+#### Example code
+
+```java
+import gears.GearsBuilder;
+import gears.readers.KeysReader;
+import gears.records.KeysReaderRecord;
+
+public class App 
+{
+    public static void main(String args[]) 
+    {  
+        // Create the reader that will pass data to the pipe
+        KeysReader reader = new KeysReader();
+        
+        // Create the data pipe builder
+        GearsBuilder<KeysReaderRecord> gb = GearsBuilder.CreateGearsBuilder(reader);
+
+        // Only process keys that start with "person:"
+        gb.filter(r->{
+        	return r.getKey().startsWith("person:");
+       	});
+
+        // Compare each person's age to the current maximum age
+        gb.foreach(r->{
+        	String newAgeStr = r.getHashVal().get("age");
+        	int newAge = Integer.parseInt(newAgeStr);
+        	
+        	// Get the current maximum age
+        	String maxAgeKey = "age:maximum";
+        	String maxAgeStr = (String) GearsBuilder.execute("GET", maxAgeKey);
+        	
+        	int maxAge = 0; // Initialize to 0
+        	if (maxAgeStr != null) {
+                // Convert the maximum age to an integer
+        		maxAge = Integer.parseInt(maxAgeStr);
+        	}
+
+        	// Update the maximum age if a new age is higher
+        	if (newAge > maxAge) {               
+        		GearsBuilder.execute("SET", maxAgeKey, newAgeStr); 
+        	}
```

Comment:
Maybe mentioned that getting maxAge and setting maxAge is not atomic and if atomicity is needed he needs to set execution mode sync on register?

Reply:
Also maybe mention that the example will not work on cluster and on cluster it is require to use `GearsBuilder.hashtag`